core/vm/evm.go

```diff
@@ -256,7 +254,6 @@ func (evm *EVM) Call(caller ContractRef, addr common.Address, input []byte, gas
 			// The depth-check is already done, and precompiles handled above
 			contract := NewContract(caller, AccountRef(addrCopy), value, gas)
 			contract.SetCallCode(&addrCopy, evm.StateDB.GetCodeHash(addrCopy), code)
-			contract.IsDeployment = creation
```

Comment:
for `creation` to be `true`, it means that the account did not exist (check L198). If the account does not exist, then `len(code) == 0` on L249. So `IsDeployment` is never `true`.

This variable is used when calling `(*EVM).create` but not `(*EVM).Call`.